components/Graphs/MetricCard.tsx

```diff
@@ -10,38 +11,42 @@ type MetricCardProps = {
 };
 
 export default function MetricCard({ stats, variant, range }: MetricCardProps) {
-  const seriesData = stats
-    ?.map((stat) => (variant === "stars" ? stat.star_count || 0 : stat.forks_count || 0))
-    .reverse();
-  const bucketData = stats?.map((stat) => new Date(stat.bucket).toDateString()).reverse();
+  const countProperty = variant === "stars" ? "star_count" : "forks_count";
+  const seriesData = useMemo(() => stats?.map((stat) => stat[countProperty] ?? 0) ?? [], [countProperty, stats]);
```

Comment:
These are things we shouldn't have to recalculate unless the props change. `useMemo` was a good fit for this. Thanks Million Lint for the assist!

Reply:
I'd love to see a Million Lint demo if you have time.

Reply:
Just got off a live steam with Toby from the team where we went through it. Happy to do a demo.

Reply:
https://github.com/open-sauced/app/issues/3104